### PR TITLE
Workflowv2 crontriggers create

### PR DESCRIPTION
**Pull request by garbage-collector**
For #1199 

https://github.com/openstack/mistral/blob/master/mistral/api/controllers/v2/cron_trigger.py#L57

### CODE REVIEW

**Comment by garbage-collector** [comment]
Ready to review!

**Comment by garbage-collector** [comment]
Need rebase.

File: openstack/workflow/v2/crontriggers/requests.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	Name string `json:"name"`
+
+	// Pattern is a Unix crontab patterns format to execute the workflow.
+	Pattern string `json:"pattern"`
+
+	// RemainingExecutions sets the number of executions for the trigger.
+	RemainingExecutions int `json:"remaining_executions,omitempty"`
+
+	// WorkflowInput is a JSON structure containing workflow input values, serialized as string.
+	WorkflowInput string `json:"workflow_input,omitempty"`
```
[the remark this answers is not in the document]
Comment (garbage-collector) [reply openstack/workflow/v2/crontriggers/requests.go]: Sending a map is working! Good catch :)

File: openstack/workflow/v2/crontriggers/requests.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import "github.com/gophercloud/gophercloud"
+
+// CreateOptsBuilder allows extension to add additional parameters to the Create request.
+type CreateOptsBuilder interface {
+	ToCronTriggerCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts specifies parameters used to create a cron trigger.
+type CreateOpts struct {
```
[the remark this answers is not in the document]
Comment (garbage-collector) [reply openstack/workflow/v2/crontriggers/requests.go]: Missing field added

File: openstack/workflow/v2/crontriggers/results.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	Scope string `json:"scope"`
+
+	// WorkflowID is the ID of the workflow linked to the trigger.
+	WorkflowID string `json:"workflow_id"`
+
+	// WorkflowName is the name of the workflow linked to the trigger.
+	WorkflowName string `json:"workflow_name"`
+
+	// WorkflowInput contains the workflow input values in a JSON stringified object.
+	WorkflowInput string `json:"workflow_input"`
```
[the remark this answers is not in the document]
Comment (garbage-collector) [reply openstack/workflow/v2/crontriggers/results.go]: It returns a JSON embedded string, but I worked to set a map[string]interface{} for input && params parameters.

File: openstack/workflow/v2/crontriggers/results.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+	// WorkflowParams contains workflow type specific parameters in a JSON stringified object.
+	WorkflowParams string `json:"workflow_params"`
+
+	// CreatedAt contains the cron trigger creation date.
+	CreatedAt time.Time `json:"created_at"`
```
[the remark this answers is not in the document]
Comment (garbage-collector) [reply openstack/workflow/v2/crontriggers/results.go]: Fixed

**Comment by garbage-collector** [comment]
New commit amend to this PR, ready to review again :)

File: openstack/workflow/v2/crontriggers/doc.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+Example to create a cron trigger
+
+	createOpts := &crontriggers.CreateOpts{
+		WorkflowID:     "w1",
+		WorkflowInput:  "{}",
+		WorkflowParams: "{}",
+		Name:           "trigger",
+	}
```
Comment (jtopjian) [inline openstack/workflow/v2/crontriggers/doc.go]: This example will need updated to reflect passing a `map[string]interface{}`

Comment (garbage-collector) [reply openstack/workflow/v2/crontriggers/doc.go]: Fixed

**Comment by garbage-collector** [comment]
Everything fixed regarding your comments, ready to review now!

**Comment by garbage-collector** [comment]
Regarding the fixtures, I'll fix this for the next PR for cron triggers (the GET one).

**Pull request merged**